src/components/common/CommentView.js

```diff
@@ -387,9 +404,99 @@ class CommentEditor extends Component {
     );
   }
 
+  renderReplyOptions() {
+    const { t, onReply, comment, classes, onDeleteThread } = this.props;
+    const { openQuickReplyMenu, anchorEl } = this.state;
+    const { options = null } = comment.data;
+
+    let replyButtons = null;
+    let replyMenu = null;
+
+    if (options) {
+      replyButtons = options
+        .slice(0, MAX_QUICK_REPLIES_TO_SHOW)
+        .map((optText) => (
+          <Button
+            className={classes.quickReplyButtons}
+            variant="outlined"
+            color="primary"
+            onClick={() => this.handleOnQuickReply(optText)}
+          >
+            {optText}
+          </Button>
+        ));
+
+      if (options.length > MAX_QUICK_REPLIES_TO_SHOW) {
+        replyMenu = (
+          <>
+            <IconButton
+              className={classes.moreQuickReplyMenu}
+              onClick={(e) => this.handleOnClickQuickReplyMenu(e)}
+            >
+              <MoreVertRounded />
+            </IconButton>
+            <Menu
+              open={openQuickReplyMenu}
+              anchorEl={anchorEl}
+              onClose={() => this.handleOnCloseQuickReplyMenu()}
+            >
+              {options
+                .slice(MAX_QUICK_REPLIES_TO_SHOW, options.length)
+                .map((optText) => (
+                  <MenuItem
+                    key={optText}
+                    onClick={() => this.handleOnQuickReply(optText)}
+                  >
+                    {optText}
+                  </MenuItem>
+                ))}
+            </Menu>
+          </>
+        );
+      }
+    }
+
+    return (
+      <Grid
+        className={classes.replyContainer}
+        container
+        spacing={1}
+        justifyContent="space-between"
+      >
+        <Grid item xs="auto">
+          {
+            // if the comment is not the end
+            _.isUndefined(comment.data.end) ? (
+              <TextField
+                className={classes.replyBox}
+                size="small"
+                variant="outlined"
+                placeholder={t('Reply ...')}
+                fullWidth
+                onClick={onReply}
+              />
+            ) : (
+              <Button
+                color="secondary"
+                variant="outlined"
+                onClick={() => onDeleteThread(comment._id)}
+              >
+                {t('Restart interaction')}
+              </Button>
+            )
+          }
+        </Grid>
+        <Grid item xs="auto">
+          {replyButtons}
+          {replyMenu}
```

Comment:
Here we can also add the logic to only show the reply box and the buttons only if it's the last comment in the thread.

Reply:
This is taken care of by line 557 where the function `renderReplyOptions()` is called only if we `showReply` which is true only when the comment has no children and the context permits it (see `CodeReview` component)